### ISSUE TITLE
Errors reported from non-CSS blocks

### ISSUE THREAD

**Issue by sathomas** [stylelint/stylelint#3219]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Errors are being reported from text outside of CSS blocks

> Which rule, if any, is this issue related to?

`string-no-newline`

> What CSS is needed to reproduce this issue?

Following `Test.vue` file:

```html
<docs></docs>
<template>
  <span
    class="
      className
    "
  />
</template>
<script></script>
<style>
.selector {
  --property: value;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

Following `.stylelintrc` file:

```json
{
  "extends": ["stylelint-config-standard"],
  "rules": {
    "indentation": "tab",
  }
}
```

> Which version of stylelint are you using?

`9.1.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "Test.vue`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Same result as version `9.1.2`, i.e. No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
 4:10  ✖  Unexpected newline in string   string-no-newline
 5:13  ✖  Unexpected newline in string   string-no-newline
```

**Comment by sathomas** [comment]
Note: potential workaround of using a processor to filter out non-css blocks does not work because of #3218

**Comment by richardtallent** [comment]
I'm seeing the same thing in my Vue templates and and MD files, with the same config -- standard + using tabs. When I remove the indentation override rule (without editing the source code), the error goes away.

**Comment by jeddy3** [comment]
Tracking this in https://github.com/stylelint/stylelint/issues/3226.

There's a pending PR to fix this.

As a workaround until the fix is released, you can turn off the `string-no-newline` rule:

```
{
  "extends": ["stylelint-config-standard"],
  "rules": {
    "indentation": "tab",
    "string-no-newline": null 
  }
}
```
